### PR TITLE
[OSDEV-788] API user. Incorrect response upon attempting to check for a match.

### PR DESCRIPTION
**Pull request by roman-stolar**
[OSDEV-788](https://opensupplyhub.atlassian.net/browse/OSDEV-788) API user. Incorrect response upon attempting to check for a match.
- Re-written logic for New_Facility/Automatic_Match when we collect & save data for FacilityListItemTemp/FacilityMatchTemp.

### CODE REVIEW

**Comment by roman-stolar** [comment]
Working on possible tests

**Comment by roman-stolar** [comment]
Team please skip Dedupe Hub App | Unittest test suite - Code coverage report as it didn't showed properly in case of missed init.py files all over the project.

File: src/dedupe-hub/api/tests/test_cumulative_matcher.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            {
+                "facility_id": "CN20241096SFEBA",
+                "confidence": 1.0,
+                "facility_list_item_id": "1",
+                "status": "AUTOMATIC",
+                "results": {}
+            },
+        ]
+        result_two = c_m.clean_matched(self.messy, processed_matches_two)
+        self.assertEqual(result_two, {
```
Comment (VadimKovalenkoSNF) [inline src/dedupe-hub/api/tests/test_cumulative_matcher.py:102]: Use `assertDictEqual` here instead.

**Review by VadimKovalenkoSNF — CHANGES_REQUESTED** [review]
Left minor comment.

File: src/dedupe-hub/api/tests/test_cumulative_matcher.py (hunk 1 above)
Comment (roman-stolar) [reply src/dedupe-hub/api/tests/test_cumulative_matcher.py]: Done

**Review by VadimKovalenkoSNF — APPROVED** [review]
Looks good overall, but need other developers to verify.

**Comment by roman-stolar** [comment]
> But pls, increase codecov issue for DD.

As I mention it an global issue that init.py files missed, so the DD coverage result are wrong.

**Comment by vladsha-dev** [comment]
> > But pls, increase codecov issue for DD.
> 
> As I mention it an global issue that init.py files missed, so the DD coverage result are wrong.

I suspect the code coverage has decreased because some files were not covered by tests previously, but now they are. It looks like these files weren't fully covered while writing the new tests, which is why it says that the code coverage decreased. So, it requires writing more tests for these specific files, or you can add `__init__.py` files in a separate PR but make this PR dependent on the first one. This way, we will not have this problem in the future with the DD hub, at least.

**Comment by roman-stolar** [comment]
> I suspect the code coverage has decreased because some files were not covered by tests previously, but now they are. It looks like these files weren't fully covered while writing the new tests, which is why it says that the code coverage decreased. So, it requires writing more tests for these specific files, or you can add **init**.py files in a separate PR but make this PR dependent on the first one. This way, we will not have this problem in the future with the DD hub, at least.

Yes, I will create seperate PR with init.py files for DD and ask to merge them first.
cc @vladsha-dev @VadimKovalenkoSNF

**Comment by roman-stolar** [comment]
@VadimKovalenkoSNF @vladsha-dev Please review & approve https://github.com/opensupplyhub/open-supply-hub/pull/216

**Review by VadimKovalenkoSNF — DISMISSED** [review]
Approved

**Review by vladsha-dev — APPROVED** [review]
LGTM

**Review by VadimKovalenkoSNF — APPROVED** [review]
Lgtm

**Pull request merged**